lib/data/entity.js

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const csv = require('csv-stringify');
+const hparser = require('htmlparser2');
+const { Readable, Transform } = require('stream');
+const { PartialPipe } = require('../util/stream');
+const { SchemaStack } = require('./schema');
+const { noop } = require('../util/util');
+
+////////////////////////////
+// ENTITY PARSING
+//
+// Parse submission XML and use with form field -> entity property mappings to build entities.
+
+const submissionXmlToEntityStream = (fields, xml) => {
```

Comment:
this function is quite similar to `submissionXmlToFieldStream`, only `opentag` is different, can we refactor to reduce duplication?